src/mappings/token.ts

```diff
@@ -59,7 +62,71 @@ export async function handleTokenDeposited(ctx: EventHandlerContext) {
         }
     }
 
-    if (!event || event?.currencyId.__kind !== 'ZenlinkLPToken') return
+    if (!event) return
+
+    let currencyId = ''
+    switch (event.currencyId.__kind) {
+        case 'ZenlinkLPToken': {
+            currencyId =
+                'ZenlinkLPToken(' + String(event.currencyId.value) + ')'
+            break
+        }
+        case 'Native': {
+            currencyId = 'Native'
+            break
+        }
+        case 'Stellar': {
+            switch (event.currencyId.value.__kind) {
+                case 'StellarNative': {
+                    currencyId = 'StellarNative'
+                    break
+                }
+                case 'AlphaNum4': {
+                    currencyId =
+                        'Stellar::AlphaNum4(' +
+                        String(event.currencyId.value.code) +
+                        ')'
+                    break
+                }
+                case 'AlphaNum12': {
+                    currencyId =
+                        'Stellar::AlphaNum12(' +
+                        String(event.currencyId.value.code) +
+                        ')'
+                    break
+                }
+            }
+        }
+        case 'XCM': {
+            switch (typeof event.currencyId.value) {
+                case 'number': {
+                    currencyId = 'XCM(' + String(event.currencyId.value) + ')'
+                    break
+                }
+                case 'object': {
+                    currencyId =
+                        'XCM(' + String(event.currencyId.value.__kind) + ')'
+                    break
+                }
+            }
+            break
+        }
+    }
```

Comment:
Let's put this logic into an extra function and re-use that function to deduplicate code.
Maybe call it `function currencyIdToString(currencyId: <the type>)` and then pass the `event.currencyId` into it. Not sure what `event.currencyId` has as a type but in case of doubt we can also use `any`.

Reply:
Changed! Thanks for the comment @ebma